src/objects/zcl_abapgit_objects_program.clas.abap

```diff
@@ -796,13 +807,22 @@ CLASS zcl_abapgit_objects_program IMPLEMENTATION.
     ls_tr_key-sub_type = 'CUAD'.
     ls_tr_key-sub_name = iv_program_name.
 
+
+    IF is_cua-adm IS NOT INITIAL.
+      ASSIGN is_cua-adm TO <ls_adm>.
+    ELSE.
+      " issue #1807 automatic correction of CUA interfaces saved incorrectly in the past (ADM was not saved in the XML)
+      CLEAR ls_adm.
+      auto_correct_cua_adm( EXPORTING is_cua = is_cua CHANGING cs_adm = ls_adm ).
+      ASSIGN ls_adm TO <ls_adm>.
+    ENDIF.
```

Comment:
Idea was to reduce the amount of code in the method, suggest having

```abap
ls_adm = is_cua-adm.
auto_correct_cua_adm( EXPORTING is_cua = is_cua CHANGING cs_adm = ls_adm ).
```

intead of the 7 statements above